Sources/ProjectDescription/CopyFilesAction.swift

```diff
@@ -0,0 +1,33 @@
+import Foundation
+
+public struct CopyFilesAction: Equatable, Codable {
+    public var name: String
+    public var destination: Destination
+    public var subpath: String
+    public var files: [FileElement]
+
+    public enum Destination: String, Codable, Equatable {
+        case absolutePath
+        case productsDirectory
+        case wrapper
+        case executables
+        case resources
+        case javaResources
+        case frameworks
+        case sharedFrameworks
+        case sharedSupport
+        case plugins
+        case other
+    }
+
+    public init(name: String,
```

Comment:
I'd keep this method private, and add a static initializer:

```swift
let target = Target(copyFiles: .plugins(name: "Copy X", files: [])
let target = Target(copyFiles: .frameworks(name: "Copy X", files: [])

// vs
let target = Target(copyFiles: .init(name: "Copy X", destination: .plugins, subpath: ""))
let target = Target(copyFiles: CopyFilesAction(name: "Copy X", destination: .plugins, subpath: ""))
```
Notice how the first one removes the redundancy that `copyFiles: CopyFilesActions` would introduce and I think it makes the API more expressive.